Update UG with find function, update various DG sections (#24)
AY2122S1-CS2103-F10-1/tp: docs/DeveloperGuide.md

| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
| `* * *`  | general user                               | add a new applicant under a position      | store his information within the system                                |
| `* * *`  | general user                               | delete an applicant from under a position | Remove applicants that are no longer related to this position          |
| `* * *`  | new user                                   | see usage instructions         | refer to instructions when I forget how to use the App                 |
| `* * *`  | user                                       | add a new person               |                                                                        |
| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |

> | `* * *`  | user                                       | easily search all relevant applicants through keywords         | I can compare applicants' relevant information |

Reviewer: ```suggestion
| `* * *`  | user                                       | easily search all relevant applicants through keywords         | compare applicants' relevant information |
```
'I can' is already at the header of the table.

| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |

> | `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |

Reviewer: ```suggestion
| `*`      | user with many applicants in the address book | sort applicants by name           | locate an applicant easily                                                 |
```

    * 3a1. MrTechRecruiter shows an error message.

      Use case resumes at step 2.

**Use case: Search for applicants through keywords**

**MSS**

> 1. User enters the keywords to search by

Reviewer: ```suggestion
1. User enters the keywords for MrTechRecruiter to search for.
```
All use case statements need full stops/periods at the end. Comment applied for all subsequent sentences.

Author: Will fix, thanks!

2. MrTechRecruiter performs the search according to the keywords
3. MrTechRecruiter displays the search results

   Use case ends.

**Extensions**

* 2a. The list is empty.

>   Use case ends.

Reviewer: The line 'use case ends' needs to be same indentation as the start of the sentence.
```suggestion
        Use case ends.
```

* 1a. An invalid flag is specified.

>     * 1a1. AddressBook shows an error message.

Author: Will fix, thanks!

>       Use case ends.

Reviewer: Same situation here. Also extension 1a should come before 2a.

Author: Will fix, thanks!

*{More to be added}*

### Non-Functional Requirements

1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
4. Should be portable and work without requiring an installer.
5. Should not depend on a remote server.

> 6. Should work for standard screen resolutions and higher.

Reviewer 2: Maybe the `portable` here could be ambiguous.
And `standard screen resolutions` can be specified to `industry-standard Full HD 1080p resolution`?

Author: Will fix, thanks!
